core/src/main/java/org/apache/iceberg/CatalogProperties.java

```diff
@@ -81,4 +81,8 @@ private CatalogProperties() {
   public static final String APP_ID = "app-id";
   public static final String USER = "user";
 
+  public static final String IO_METRICS_IMPL = "io.metrics-impl";
+  public static final String IO_METRICS_IMPL_DEFAULT = "org.apache.iceberg.hadoop.HadoopMetricsContext";
+  public static final String IO_METRICS_SCHEME = "io.metrics-scheme";
```

Comment:
`io.metrics-scheme` is hadoop specific since it's tied to a hadoop filesystem scheme in terms of reporting.  Do we really need to move it here?

Reply:
I think there still needs to be a way for users to input some sort of grouping concept in the metrics, for example users who use `S3FileIO` for `gs://` should expect metrics to land to some different metrics namespace from `s3://`. Scheme seems to be a good way to do that. For our CloudWatch metrics integration. we will match this to the CloudWatch namespace concept, where the metrics will by default be emitted to a namepace `iceberg/<scheme>` such as `iceberg/s3`

Reply:
Maybe we can give a more generic name for this than `scheme`, something like `namespace`? @kunal0829 what is the terminology used in statsd and collectd for this?

Reply:
Both statsd and prometheus it is called namespace. Collectd takes a combination of host and plugin to create a "namespace".

Reply:
sounds like using a term `io.metrics.namespace` seems like a more generic solution? @danielcweeks